xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsEventExecutor.java

```diff
@@ -131,16 +137,17 @@ public void dispose()
     @Override
     public void execute(DocumentReference documentReference, DocumentReference authorReference, String version)
     {
+        MentionsData data = new MentionsData()
+                                .setDocumentReference(documentReference.toString())
```

Comment:
`toString()`` should be used only when you need to "print" the reference (like in a log message). Otherwise, if some other code may be resolving the string reference later then you should use a proper reference serializer.

Reply:
Replaced by `EntityReferenceSerializer<String>`